sklearn/datasets/_svmlight_format.pyx

```diff
@@ -52,6 +53,12 @@ def _load_svmlight_file(f, dtype, bint multilabel, bint zero_based,
     else:
         labels = array.array("d")
 
+    if offset > 0:
+        f.seek(offset)
+        # drop the previous line that might be truncated and is to be fetched
```

Comment:
Is this not dropping the **current** line? Also I don't really understand the "is to be fetched by another call"